lib/cisco_node_utils/vxlan_global.rb

```diff
@@ -117,15 +152,19 @@ def anycast_gateway_mac
       config_get('vxlan_global', 'anycast_gateway_mac')
     end
 
-    def anycast_gateway_mac_set(configure, mac_addr)
+    def anycast_gateway_mac=(mac_addr)
       fail TypeError unless mac_addr.is_a?(String)
-      if configure == 'True'
+      if mac_addr == default_anycast_gateway_mac
         config_set('vxlan_global', 'anycast_gateway_mac',
-                   state: '', mac_addr: mac_addr)
+                   state: 'no', mac_addr: '')
       else
         config_set('vxlan_global', 'anycast_gateway_mac',
-                   state: 'no', mac_addr: '')
+                   state: '', mac_addr: mac_addr)
       end
```

Comment:
This would be cleaner if you did:

```
    if mac_addr == default_anycast_gateway_mac
      state = 'no'
      mac_addr = ''
    else
      state = ''
    end
    config_set('vxlan_global', 'anycast_gateway_mac',
                   state: state, mac_addr: mac_addr)
```